packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -577,22 +578,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
      *
      * @param widget widget file resource of which should be revealed and selected
      */
-    async selectWidgetFileNode(widget: Widget | undefined): Promise<void> {
+    async selectWidgetFileNode(widget: Widget | undefined, isSelectedByEditor?: boolean): Promise<void> {
         if (Navigatable.is(widget)) {
             const resourceUri = widget.getResourceUri();
             if (resourceUri) {
                 const { model } = await this.widget;
                 const node = await model.revealFile(resourceUri);
                 if (SelectableTreeNode.is(node)) {
-                    model.selectNode(node);
+                    model.selectNode(node, isSelectedByEditor ? true : false);
                 }
             }
         }
     }
 
-    protected onCurrentWidgetChangedHandler(): void {
+    protected onCurrentWidgetChangedHandler(args: FocusTracker.IChangedArgs<Widget>): void {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const isSelectedByEditor = (args.oldValue as any).editor && (args.newValue as any).editor;
```

Comment:
> It's probably best to avoid these `as any` casts if possible. One possibility is to use `NavigatableWidget.is()` to identify something that is (very probably) an editor without having to refer to any editor-specific fields.

done.